linetools/analysis/linelimits.py

```diff
@@ -0,0 +1,147 @@
+        """
+        Parameters
+        ----------
+        wrest : Quantity
+          Rest wavelength of the line.  Should match line.wrest
+        z : float
+          Redshift of the line.  Should match line.attrib['z']
+        zlim : tuple or list
+          Redshift limits for a line
+          Defined as wave/wrest - 1.
+        """
+        # Error checking
+        if not isinstance(z, float):
+            raise IOError("Input z must be a float")
+        if not isinstance(zlim, (tuple,list)):
+            raise IOError("Input zlim must be a tuple or list")
+        if not isinstance(wrest, Quantity):
+            raise IOError("Input wrest must be a quantity")
+        # Data
+        self._data = {}
+        # Set
+        self._z = z
+        self._wrest = wrest
+        self.set(zlim)
+
+    @property
+    def zlim(self):
+        """ Return zlim
+        """
+        return self._data['zlim']
+
+    @property
+    def wvlim(self):
+        """ Return wvlim
+        """
+        return self._data['wvlim']
+
+    @property
+    def vlim(self):
+        """ Return vlim
+        """
+        return self._data['vlim']
+
+    def reset(self):
+        """ Update all the values
+        """
+        self._data['zlim'] = self._zlim
+        self._data['wvlim'] = self._wrest*(1+np.array(self._zlim))
+        self._data['vlim'] = ltu.give_dv(self._zlim, self._z)
+        #self._data['vlim'] = ckms*((self._data['wvlim']-self._wrest*(1+self._z))/(
+        #    self._wrest*(1+self._z))).decompose()
+
+    def set(self, inp):#, itype='zlim'):
+        """ Over-ride = to re-init values
+
+        Parameters
+        ----------
+        inp : tuple, list, or Quantity array
+          * If floats -> zlim : Redshift limits
+          * If Quantity array with length units  -> wvlim : Wavelength limits
+          * If Quantity array with speed units  -> vlim : Velocity limits
+        itype : str, optional
+          Input type
+
+
+        Returns
+        -------
+
+        """
+        # Checks
+        if not isinstance(inp, (tuple, list, Quantity)):
+            raise IOError("Input must be tuple, list or Quantity")
+        '''
+        if itype == 'zlim':
+            self._data['zlim'] = inp
+        else:
+            raise IOError("Input type must be zlim, vlim, or wvlim")
+        '''
+        if isinstance(inp[0], float):  # zlim
+            self._zlim = inp
+        elif isinstance(inp[0], Quantity):  # wvlim or vlim
+            try:  # wvlim
+                self._zlim = (inp/self._wrest).decompose().to(
+                        u.dimensionless_unscaled).value - 1.
+            except UnitConversionError:
+                try:
+                    self._zlim = ltu.give_dz(inp, self._z)
+                except UnitConversionError:
+                    raise IOError("Quantity must be length or speed")
+        else:
+            raise IOError("Input must be floats or Quantities")
+        # Reset
```

Comment:
Please add inline comment saying that `self._zlim` has been updated to match the given input before applying reset (for future developers).